src/ray/core_worker/reference_count.cc

```diff
@@ -462,6 +462,13 @@ void ReferenceCounter::DeleteReferenceInternal(ReferenceTable::iterator it,
       ReleaseLineageReferencesInternal(ids_to_release);
     }
 
+    if (local_raylet_client_) {
+      if (it->second.owned_by_us) {
+        local_raylet_client_->DeleteSpilledObject(
+            id, [](const Status &status, const rpc::DeleteSpilledObjectReply &reply) {});
+      }
+    }
```

Comment:
Hmm what if we deleted spilled objects directly from the owner? It can be done in a background thread to avoid blocking other operations.